### PR TITLE
Infrastructure/3133 update latest google charts  v51 upgrade

### PR DESCRIPTION
**Pull request by danielgent**
## Summary

Addresses issue #3133

## Checklist

- [ ] My code is tested and passes existing unit tests.
- [ ] My code has an appropriate set of unit tests which all pass.
- [ ] My code is backward-compatible with WordPress 4.7 and PHP 5.6.
- [ ] My code follows the [WordPress](https://make.wordpress.org/core/handbook/best-practices/coding-standards/) coding standards.
- [ ] My code has proper inline documentation.
- [x] I have added a QA Brief on the issue linked above.
- [x] I have signed the Contributor License Agreement (see <https://cla.developers.google.com/>).

### CODE REVIEW

**Review by aaemnnosttv — CHANGES_REQUESTED** [review]
Looks good for the most part, just a few things we might want to change - we should keep the `chartVersion` prop to the `Chart` component though.

File: assets/js/components/GoogleChart.js [hunk 1]
```diff
@@ -194,7 +194,6 @@ export default function GoogleChart( props ) {
 				className="googlesitekit-chart__inner"
 				chartEvents={ combinedChartEvents }
 				chartType={ chartType }
-				chartVersion="49"
```
[the remark this answers is not in the document]
Comment (danielgent) [reply assets/js/components/GoogleChart.js:197]: Ok. Just noting that this step was mentioned explicitly in the IB.  I assumed a previous bug was called by the versions not being matched.. I'll add this prop back but set to the latest version :+1:

File: includes/Core/Assets/Assets.php [hunk 2]
```diff
@@ -436,7 +436,7 @@ private function get_assets() {
 					'src'          => 'https://www.gstatic.com/charts/loader.js',
 					'in_footer'    => false,
 					'before_print' => function( $handle ) {
-						wp_add_inline_script( $handle, 'google.charts.load( "49", { packages: [ "corechart" ] } );' );
+						wp_add_inline_script( $handle, 'google.charts.load( "51", { packages: [ "corechart" ] } );' );
```
[the remark this answers is not in the document]
Comment (felixarntz) [reply includes/Core/Assets/Assets.php:439]: @aaemnnosttv Are we using the new `Chart` component consistently throughout now? I don't recall, but we've probably removed all usages of the old way - in that case I agree, we should get rid of the manual loading calls here and in Storybook. Good point on adding preloading still, let's do that for the Site Kit admin screens at least.

File: .storybook/preview-head.html [hunk 3]
```diff
@@ -29,7 +29,7 @@
 </script>
 <script src="https://www.gstatic.com/charts/loader.js"></script>
 <script>
-	google.charts.load( '49', {
+	google.charts.load( '51', {
```
[the remark this answers is not in the document]
Comment (danielgent) [reply .storybook/preview-head.html]: I removed this and all the storybook tests pass so this looks good to go :+1:

File: includes/Core/Assets/Assets.php (hunk 2 above)
[the remark this answers is not in the document]
Comment (danielgent) [reply includes/Core/Assets/Assets.php]: I've just removed this whole enqueue section and the e2e tests work *BUT* that's not exactly what change is being asked for with _"preloading"_ :thinking: @aaemnnosttv

[the remark this answers is not in the document]
Comment (aaemnnosttv) [reply includes/Core/Assets/Assets.php]: @felixarntz we only have the one `GoogleChart` component for a while now which uses the React `Chart` within so unless we're manually creating a chart somewhere then it should be all using the same component. I can't think of any exceptions to this so I think we should be safe.

```
8 results - 8 files

assets/js/components/MiniChart.js:
  74: 			<GoogleChart

assets/js/components/Sparkline.js:
  73: 			<GoogleChart

assets/js/modules/adsense/components/module/ModuleOverviewWidget/Stats.js:
  135: 					<GoogleChart

assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserCountGraph.js:
  188: 			<GoogleChart

assets/js/modules/analytics/components/dashboard/DashboardAllTrafficWidget/UserDimensionsPieChart.js:
  471: 				<GoogleChart

assets/js/modules/analytics/components/module/ModuleAcquisitionChannelsWidget/PieChart.js:
  37: 			<GoogleChart

assets/js/modules/analytics/components/module/ModuleOverviewWidget/SiteStats.js:
  84: 					<GoogleChart

assets/js/modules/search-console/components/module/ModuleOverviewWidget/Stats.js:
  131: 					<GoogleChart
```

As for preloading, this is not the right term (I always forget which one) but looking at [`wp_resource_hints`](https://developer.wordpress.org/reference/functions/wp_resource_hints/) which is what core uses, `wp_dependencies_unique_hosts` is what grabs all the hosts from enqueued styles and scripts so to preserve this, we'd only need to hook in to add `www.gstatic.com` to the list of `dns-prefetch`.

[the remark this answers is not in the document]
Comment (felixarntz) [reply includes/Core/Assets/Assets.php]: Thanks @aaemnnosttv, both makes sense.

[the remark this answers is not in the document]
Comment (aaemnnosttv) [reply includes/Core/Assets/Assets.php]: @felixarntz after digging into this a bit, it seems `wp_dependencies_unique_hosts` does not traverse dependencies, only top-level scripts and styles (those "directly" enqueued) so this dns-prefetch isn't something we currently have now even when the chart loader is printed on the page.

I'll open a new issue for us to explore this in the future but we don't need to block this one to preserve something we aren't losing 😄

File: .storybook/preview-head.html [hunk 4]
```diff
@@ -28,8 +28,3 @@
 	} );
 </script>
 <script src="https://www.gstatic.com/charts/loader.js"></script>
```
Comment (aaemnnosttv) [inline .storybook/preview-head.html:30]:
```suggestion
```

**Review by aaemnnosttv — COMMENTED** [review]
LGTM, just the loader to remove as well.

**Comment by aaemnnosttv** [comment]
@googlebot I consent.

**Comment by aaemnnosttv** [comment]
VRT failure was usual unstable Buttons test.

**Pull request merged**